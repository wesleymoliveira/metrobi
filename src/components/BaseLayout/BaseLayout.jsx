import React from 'react';
import { Footer, Header } from '../';

import * as S from './BaseLayout.styled';

export function BaseLayout({ children }) {
	return (
    <S.Wrapper data-testid='base-layout'>
			<Header />
      <S.Content>{children}</S.Content>
      <Footer /> 
    </S.Wrapper>
  );
}
