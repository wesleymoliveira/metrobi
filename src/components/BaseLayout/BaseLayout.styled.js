import styled, {css} from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    height: 100vh;
    gap: ${theme.spacings.gap};
    //max-width: ${theme.flex.container};
  `}
`;

export const Content = styled.main`
  ${({theme}) => css`
    flex: 8;
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.gap};
  `}
`;
