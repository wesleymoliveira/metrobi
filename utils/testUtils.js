import React from "react";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import theme from "../src/styles/theme";

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options });
};

const snapshotRenderer = (component) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
};

export * from "@testing-library/react";

export { customRender as render, snapshotRenderer  };
