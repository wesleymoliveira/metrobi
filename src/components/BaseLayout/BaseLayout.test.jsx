import { screen } from '@testing-library/react';

import { BaseLayout } from './BaseLayout';
import { render, snapshotRenderer } from '../../../utils/testUtils';

describe('<BaseLayout />', () => {
	it('should render the component', () => {
		render(<BaseLayout />);
		
		expect(screen.getByTestId(/base-layout/i)).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const tree = snapshotRenderer(<BaseLayout />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
