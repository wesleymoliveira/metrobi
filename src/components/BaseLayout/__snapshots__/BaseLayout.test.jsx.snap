// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseLayout /> should match snapshot 1`] = `
.c2 {
  color: #FAFAFA;
  font-size: 2.8rem;
}

.c1 {
  background-color: #54d5eb;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  background-color: #FFA500;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 100vh;
  gap: 1.0rem;
}

.c3 {
  -webkit-flex: 8;
  -ms-flex: 8;
  flex: 8;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.0rem;
}

<div
  className="c0"
  data-testid="base-layout"
>
  <header
    className="c1"
    data-testid="header"
  >
    <h1
      className="c2"
      color="white"
      data-testid="typography"
      size="xxlarge"
    >
      Header
    </h1>
  </header>
  <main
    className="c3"
  />
  <footer
    className="c4"
    data-testid="footer"
  >
    <h1
      className="c2"
      color="white"
      data-testid="typography"
      size="xxlarge"
    >
      Footer
    </h1>
  </footer>
</div>
`;
